### PR TITLE
add num-utils package

### PR DESCRIPTION
**Pull request by matthiasdiener**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
This is ready for review @conda-forge/help-perl @conda-forge/staged-recipes

File: recipes/num-utils/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+requirements:
+  build:
+    - perl
+  run:
+    - perl
+
+test:
+  commands:
+    - $PREFIX/bin/numrange /1..10/
```
Comment (chrisburr) [inline recipes/num-utils/meta.yaml:38]:
```suggestion
    - numrange /1..10/
```

File: recipes/num-utils/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - patches/10_help_locks_stdin.patch
+    - patches/11_upstream_URI_manp.patch
+
+build:
+  noarch: generic
+  number: 0
+
+requirements:
+  build:
+    - perl
```
Comment (chrisburr) [inline recipes/num-utils/meta.yaml:32]:
```suggestion
    - perl
    - sed
```

**Comment by matthiasdiener** [comment]
Thank you for the comments @chrisburr ! This is ready for another review.

**Pull request merged**